transformer_nuggets/llama/finetune.py

```diff
@@ -19,6 +20,7 @@
 import torch.multiprocessing as mp
 import transformer_nuggets.llama.train
 import transformer_nuggets.quant.qlora as qlora
+from torch.distributed._composable.fsdp import fully_shard
```

Comment:
does this import need andrews PR to land?

Reply:
if so I think just move it it into the fsdp_2 flag and then this looks good!

Reply:
The `reshard_after_forward` arg has not landed yet (it is the current bottom of the PR stack), so this PR will not work with `main` today.

I think it would be best to wait for at least the next 2 PRs to land if not the next 4. I am still giving other some time to review.

Reply:
cool we can let this sit then

Reply:
converted to draft until FSDP2 gets landed. we can use this PR as a reference